apps/qubit/modules/object/actions/exportAction.class.php

```diff
@@ -0,0 +1,158 @@
+
+        foreach (QubitTerm::getLevelsOfDescription() as $item)
+        {
+          $this->choices[$item->id] = $item->__toString();
+        }
+
+        $size = count($this->choices);
+        if ($size === 0)
+        {
+          $size = 4;
+        }
+
+        $this->form->setWidget('levels', new sfWidgetFormSelect(array('choices' => $this->choices, 'multiple' => true), array('size' => $size)));
+
+        break;
+
+      default:
+        return parent::addField($name);
+    }
+  }
+
+  protected function processField($field)
+  {
+    switch ($field->getName())
+    {
+      case 'levels':
+        $this->levels = $this->form->getValue('levels');
+        if (empty($this->levels))
+        {
+          $this->levels = array();
+        }
+
+        break;
+    }
+  }
+
+  protected function doBackgroundExport($request)
+  {
+    // Export type, CSV or XML?
+    $this->type = $request->getParameter('type', 'csv');
+
+    // Create array of selections to pass to background job where Term ID will
+    // be key, and Term description is value.
+    foreach ($this->levels as $value)
+    {
+      $levelsOfDescription[$value] = $this->choices[$value];
+    }
+
+    $options = array('params' => array('fromClipboard' => true,
+                                       'slugs' => $this->context->user->getClipboard()->getAll()),
+                     'include-all-levels' => ('on' == $request->getParameter('includeAllLevels')) ? true : false,
+                     'current-level-only' => ('on' == $request->getParameter('includeDescendants')) ? false : true,
+                     'public' => ($request->getParameter('includeDrafts') == 'on') ? false : true,
+                     'objectType' => $request->getParameter('objectType'),
+                     'levels' => $levelsOfDescription);
+
+    try
+    {
+      if ('CSV' == strtoupper($this->type))
+      {
+        QubitJob::runJob('arInformationObjectCsvExportJob', $options);
+      }
+      else
+      {
+        QubitJob::runJob('arXmlExportJob', $options);
+      }
+
+      // Let user know export has started
+      sfContext::getInstance()->getConfiguration()->loadHelpers(array('Url'));
+
+      $message = $this->context->i18n->__('%1%Export of descriptions initiated.%2% Check %3%job management%4% page to download the results when it has completed.', array(
+        '%1%' => '<strong>',
+        '%2%' => '</strong>',
+        '%3%' => sprintf('<a href="%s">', url_for(array('module' => 'jobs', 'action' => 'browse'))),
+        '%4%' => '</a>'));
+
+      $this->context->user->setFlash('notice', $message);
+    }
+    catch (sfException $e)
+    {
+      $this->context->user->setFlash('error', $e->getMessage());
+      return sfView::SUCCESS;
+    }
+  }
+
+  public function execute($request)
+  {
+    parent::execute($request);
+
+    $this->response->addJavaScript('exportOptions', 'last');
+
+    if (isset($request->type))
+    {
+      $this->type = $request->type;
+    }
```

Comment:
I think a default value is needed in `$this->type` for GET requests, as it's being used in the template in the type field. I'd replace this if with they way it's being done in the doBackgroundExport function:

`$this->type = $request->getParameter('type', 'csv');`